### ISSUE TITLE
[Clock] time values are not strict monotonically increasing

### ISSUE THREAD

**Issue by OpenModelica-TracImporter** [OpenModelica/OpenModelica#3759]
The following simple test case produces a result file with time values that are not strict monotonically increasing.

```modelica
setCommandLineOptions("+std=3.3"); getErrorString();
loadString("
model TestClock
  Integer nextInterval(start=1);
  Clock c = Clock(nextInterval, 10);
equation
  when c then
    nextInterval = previous(nextInterval) + 1;
  end when;
end TestClock;
"); getErrorString();
simulate(TestClock, outputFormat="csv"); getErrorString();
```

---------
Taken from https://trac.openmodelica.org/OpenModelica/ticket/3759

**Comment by AnHeuermann** [comment]
Should be resolved during fixing #7854

**Comment by AnHeuermann** [comment]
Fixed in #8287
